src/views/ModelPlan/TaskList/Basics/Milestones/index.tsx

```diff
@@ -245,19 +249,30 @@ const Milestones = () => {
                       {h('datePlaceholder')}
                     </div>
                     <FieldErrorMsg>{flatErrors.completeICIP}</FieldErrorMsg>
-                    <Field
-                      as={DatePicker}
-                      error={+!!flatErrors.completeICIP}
-                      id="Milestone-completeICIP"
-                      maxLength={50}
-                      name="completeICIP"
-                      defaultValue={completeICIP}
-                      onBlur={(e: any) =>
-                        handleOnBlur(e.target.value, 'completeICIP')
-                      }
-                    />
+                    <div className="position-relative">
+                      <Field
+                        as={DatePicker}
+                        error={+!!flatErrors.completeICIP}
+                        id="Milestone-completeICIP"
+                        maxLength={50}
+                        name="completeICIP"
+                        defaultValue={completeICIP}
+                        onBlur={(e: any) =>
+                          handleOnBlur(e.target.value, 'completeICIP')
+                        }
+                      />
+                      {isDateInPast(values.completeICIP) && (
+                        <DatePickerWarning label={h('dateWarning')} />
+                      )}
+                    </div>
```

Comment:
With how frequently this shows up, do you think it'd be worth making a component that wraps `<Field as={DatePicker}` that handles the `isDateInPast` logic? I.e. a (naming aside) `<DateFieldWarnInPast>` component?

Not requesting changes per-se, but opening up for discussion!

Reply:
Upon further review in other sections, this might get a bit hairy because there's multiple fields that map to a single warning in some spots.

Reply:
That's a good idea, I should be able to make the component flexible enough to handle these cases.  I might take a. little time to wrap this in a custom component

Reply:
Upon further inspection of the Truss DatePicker, there are only onChange and onBlur handlers.  I'm not seeing any prop for a onSelect callback.  I think I remember this was a discussion Gary and I were having with this component.